include/vapor/VaporField.h

```diff
@@ -11,6 +11,60 @@
 
 namespace flow
 {
+
+//
+// Helper class: it is used to identify a specific grid.
+//
+class FLOW_API GridKey final
+{
+private:
+    std::string varName; 
+
+    // Buffer definition:
+    // Offset   Data_Type   Information
+    //
+    // 0        uint32_t    currentTS
+    // 4        int32_t     refLevel
+    // 8        int32_t     compLevel
+    // 12       double[3]   extent_min
+    // 36       double[3]   extent_max
+    // 60       float       DefaultZ
+    //
+    // Total size: 64
```

Comment:
Using hard-coded data sizes is error prone and hard to maintain. Can't you define a struct for these quantities and make use of sizeof? Alternatively, does 'buf' need to be an std::array? 

Perhaps the bigger question is why pack the quantities into an array at all? Why not just retain and compare the individual quantities? Some additional documentation would be helpful.

Reply:
My thinking was that when comparing two buffers, I could use one single function call `std::strcmp` to compare the content of two buffers, instead of calling approximately a dozen individual comparisons. A small advantage is that it's also faster this day.

Regarding to documetation, I've clearly documented what offset stores what information of what size. Any suggestion on further documentation is welcome.

Reply:
@shaomeng please see, for example,

https://stackoverflow.com/questions/28858359/should-i-use-memcmp-or-chained-equal-to-operations-when-both-give-the-same-resul

for one discussion on how to handle class equality operators. Of course you could simply just define a member-wise equality operator, but it is certainly possible to use std::memcmp in a way that follows best programming practices (e.g. avoids having to know the size of each data element, results in more maintainable code, etc.)

Reply:
@clyne 

The discussion seems comprehensive, but I don't take anything away that I think I could use in my use case. The current implementation is using std::memcmp to directly comparing two memory buffers. This approach results in fairly readable code, and should produce the highest performance,  but the downside is that the hard coded memory offsets look ugly. In this particular case I decided that it's an acceptable price to pay. Going back to the discussion thread, maybe I'm missing something? If so, please specify.